### PR TITLE
kvm-unit-tests: Adding git reference parameter

### PR DESCRIPTION
**Pull request by nareshkamboju**
Adding git reference parameter for kvm unit tests to git checkout on
given git reference like branch/tag/commit-id

Signed-off-by: Naresh Kamboju <naresh.kamboju@linaro.org>

### CODE REVIEW

**Comment by mwasilew** [comment]
```
* SHELLCHECK: [OUTPUT]:
 
 In automated/linux/kvm-unit-tests/kvm-unit-tests.sh line 58:
       git checkout 40f559bc572e355cadef513ec41fe114e9a8394e -b ${BRANCH}
                                                                ^-------^ SC2086: Double quote to prevent globbing and word splitting.
```

File: automated/linux/kvm-unit-tests/kvm-unit-tests.sh [hunk 1]
```diff
@@ -8,16 +8,18 @@ RESULT_FILE="${OUTPUT}/result.txt"
...
+    echo "Usage: $0 [-s <true|false>] [-m <true|false>] [-g branch]" 1>&2
     exit 1
 }
 
-while getopts "s:m:h" o; do
+while getopts "s:m:g:h" o; do
   case "$o" in
     s) SKIP_INSTALL="${OPTARG}" ;;
     m) SMP="${OPTARG}" ;;
+    g) export BRANCH="${OPTARG}" ;;
```
Comment (mwasilew) [inline automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: why is there 'export' here?

**Comment by mwasilew** [comment]
Given the commit message, does this mean that the test only works on kernel 5.2?

File: automated/linux/kvm-unit-tests/kvm-unit-tests.sh [hunk 2]
```diff
@@ -48,9 +50,14 @@ kvm_unit_tests_run_test() {
 
 kvm_unit_tests_build_test() {
     info_msg "git clone kvm unit tests ..."
-    git clone https://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git --depth 1
+    git clone https://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git
     # shellcheck disable=SC2164
     cd kvm-unit-tests
+    if [ "${BRANCH}" = "5.2" ]; then
```
Comment (danrue) [inline automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: I don't understand why this is hard coded. Can we just have a GIT_REF parameter and "git checkout" it regardless of what it is? What am I missing?

File: automated/linux/kvm-unit-tests/kvm-unit-tests.sh (hunk 1 above)
Comment (nareshkamboju) [reply automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: export removed.

**Comment by nareshkamboju** [comment]
> does this mean that the test only works on kernel 5.2?

kvm-unit-tests master branch do not cope with linux-stable-5.2 and lower branches.
To get this working on linux-stable-5.2 I have to checkout on a working commit id.

**Comment by nareshkamboju** [comment]
> ```
> * SHELLCHECK: [OUTPUT]:
>  
>  In automated/linux/kvm-unit-tests/kvm-unit-tests.sh line 58:
>        git checkout 40f559bc572e355cadef513ec41fe114e9a8394e -b ${BRANCH}
>                                                                 ^-------^ SC2086: Double quote to prevent globbing and word splitting.
> ```

Resolved.

**Comment by mrchapp** [comment]
We can probably 1) rename "BRANCH" to "GIT_REF", 2) default $GIT_REF to "master", then 3) git-clone and immediately git-reset --hard to the indicated $GIT_REF.

File: automated/linux/kvm-unit-tests/kvm-unit-tests.sh (hunk 2 above)
Comment (nareshkamboju) [reply automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: @danrue Addressed your comments with GIT_REF.

File: automated/linux/kvm-unit-tests/kvm-unit-tests.yaml [hunk 3]
```diff
@@ -24,9 +24,10 @@ metadata:
 params:
     SKIP_INSTALL: "False"
     SMP: "true"
+    GIT_REF: ""
```
Comment (danrue) [inline automated/linux/kvm-unit-tests/kvm-unit-tests.yaml]: I think if you pass in "" it won't get defaulted to "master" as intended.

Comment (nareshkamboju) [reply automated/linux/kvm-unit-tests/kvm-unit-tests.yaml]: Thanks for the comments.
GIT_REF default as master now.
 GIT_REF: "master"

**Comment by nareshkamboju** [comment]
@mwasilew @danrue @mrchapp Thanks for providing review comments.
Your comments have been addressed and new version of patch is under review.
As we discussed keeping test-definitions are more generic for inside and outside of Linaro use cases.

File: automated/linux/kvm-unit-tests/kvm-unit-tests.sh [hunk 4]
```diff
@@ -48,9 +52,12 @@ kvm_unit_tests_run_test() {
 
 kvm_unit_tests_build_test() {
     info_msg "git clone kvm unit tests ..."
-    git clone https://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git --depth 1
+    git clone https://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git
     # shellcheck disable=SC2164
     cd kvm-unit-tests
+    echo "Checkout on a given git reference ${GIT_REF}"
```
Comment (mwasilew) [inline automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: please use info_msg

Comment (nareshkamboju) [reply automated/linux/kvm-unit-tests/kvm-unit-tests.sh]: @mwasilew Thanks 
Done.

**Pull request merged**